src/encodings/decoding.rs

```diff
@@ -303,6 +311,70 @@ impl<T: DataType> Decoder<T> for DictDecoder<T> {
   }
 }
 
+// ----------------------------------------------------------------------
+// RLE Decoding
+
+/// RLE decoding for values.
+/// Currently is used only for data pages v2 and supports boolean types.
+pub struct RleValueDecoder<T: DataType> {
+  values_left: usize,
+  decoder: Option<RleDecoder>,
+  _phantom: PhantomData<T>
+}
+
+impl<T: DataType> RleValueDecoder<T> {
+  pub fn new() -> Self {
+    Self { values_left: 0, decoder: None, _phantom: PhantomData }
+  }
+
+  #[inline]
+  fn internal_set(&mut self, data: ByteBufferPtr, num_values: usize) -> Result<()> {
```

Comment:
nit: can we rename this to `set_data_internal`? it has the same signature as `set_data`.